File: Interviewbit/Tree/symmetric-binary-tree.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {
    public boolean solve(TreeNode A, TreeNode B){
        if(A==null && B==null) return true;
        if(A!=null && B==null ||A==null &&B!=null) return false;
        if(A.val!=B.val) return false;
        return (solve(A.left,B.right) && solve(A.right,B.left));
    }
    
    public int isSymmetric(TreeNode A) {
        if(A==null) return 1;
        if(solve(A.left,A.right)) return 1;
        else return 0;
    }
}

File: Interviewbit/Tree/symmetric-binary-tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
 bool solve(TreeNode* A, TreeNode* B){
     if(A==NULL && B==NULL) return true;
     if(A==NULL &&B!=NULL ||A!=NULL &&B==NULL) return false;
     return (A->val==B->val)&& solve(A->left,B->right)&& solve(A->right,B->left);
 }
 
int Solution::isSymmetric(TreeNode* A) {
    if(A==NULL) return 1;
    return solve(A->left,A->right);
}
